### PR TITLE
Resolving Issue 340

### PR DESCRIPTION
**Pull request by Baalmart**
# Bug when retrieving recent events without the RECENT query parameter

**_WHAT DOES THIS PR DO?_**
Resolves[ issue-340](https://github.com/airqo-platform/AirQo-api/issues/340)

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/issue-340

**_HOW DO I TEST OUT THIS PR?_**
Using a separate terminal, start your local REDIS server accordingly.
`cd AirQo-api/src/device-registry`
`npm install`
`npm run stage-mac `OR `npm run prod-pc`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
[GET Events endpoint.](https://docs.airqo.net/airqo-platform-api/device-registry#get-events)
Use port 3000 when testing on localhost.

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A

### CODE REVIEW

**Review by ja-odur — APPROVED** [review]
LGTM, although some devices are showing a last push of 27-03-2021, not sure whether it a job/pipeline issue, actual device issue or PR logic issue (AirQo). But the kCCA results are all recent (29-04-2021). Probably we need a way to easily tell the difference.

**Review by sserurich — APPROVED** [review]
LGTM

File: src/device-registry/utils/get-measurements.js [hunk 1]
```diff
@@ -19,6 +20,60 @@ const { generateDateFormat, generateDateFormatWithoutHrs } = require("./date");
 
 const getDetail = require("../utils/get-device-details");
 
+const isRecentTrue = (recent) => {
+  let isRecentEmpty = isEmpty(recent);
+  if (isRecentEmpty == true) {
+    return true;
+  }
+  if (recent.toLowerCase() == "yes" && isRecentEmpty == false) {
```
Comment (Baalmart) [inline src/device-registry/utils/get-measurements.js:28]: Hi @ja-odur 
just reminding us that we shall have to make appropriate changes in the frontend after this gets merged into staging. Key thing to note is that recent query parameter now takes strings **yes** or **no**....something which is different from the previous expected values of either **true** or **false**.

**Comment by Baalmart** [comment]
> LGTM, although some devices are showing a last push of 27-03-2021, not sure whether it a job/pipeline issue, actual device issue or PR logic issue (AirQo). But the kCCA results are all recent (29-04-2021). Probably we need a way to easily tell the difference.

This is due to an issue with ThingSpeak,  I am informed that devices have not been streaming data since that date. The hardware team is aware and following up on this.

**Pull request merged**